src/components/TodoItem.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useContext, useState } from 'react';
+import { Todo } from '../types/Todo';
+import classNames from 'classnames';
+import { TodosContext } from './TodosContext';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { todos, setTodos } = useContext(TodosContext);
+  const [editingTodoId, setEditingTodoId] = useState<number | null>(null);
+  const [newTitle, setNewTitle] = useState(todo.title);
+
+  function handleClickDeleteTodo(TodoId: number) {
+    const newTodos = todos.filter(currTodo => currTodo.id !== TodoId);
+
+    setTodos(newTodos);
+  }
+
+  const handleEditTodo = (TodoChange: Todo) => {
+    let newTodos = todos;
+
+    if (newTitle === TodoChange.title) {
+      setEditingTodoId(null);
+
+      return;
+    }
+
+    if (newTitle.trim()) {
+      newTodos = todos.map(curr => {
+        if (curr.id === todo.id) {
+          return {
+            ...curr,
+            title: newTitle,
+          };
+        }
+
+        return curr;
+      });
+    } else {
+      handleClickDeleteTodo(todo.id);
+
+      return;
+    }
+
+    setTodos(newTodos);
+    setEditingTodoId(null);
+  };
+
+  const handleChangeChecked = (todoChange: Todo) => {
+    const newTodos = todos.map(currTodo => {
+      if (currTodo.id === todoChange.id) {
+        return {
+          ...currTodo,
+          completed: !todoChange.completed,
+        };
+      }
+
+      return currTodo;
+    });
+
+    setTodos(newTodos);
+  };
+
+  return (
+    <li
+      data-id={todo.id}
+      className={classNames({
+        completed: todo.completed,
+        editing: todo.id === editingTodoId,
+      })}
+      onDoubleClick={() => setEditingTodoId(todo.id)}
```

Comment:
also, set the current title to `newTitle`, because now it works not good:

start editing todo, type something. Then press escape - todo returned to the previous value. But when you start editing this todo again the previous typed during editing value appears